discovery-infra/test_infra/assisted_service_api.py

```diff
@@ -210,13 +230,18 @@ def download_infraenv_image(self, infraenv_id: str, image_path: str) -> None:
         response_obj = response[0]
         self._download(response=response_obj, file_path=image_path, verify_file_size=True)
 
+#TODO remove and use infraenv
     def update_hosts(
             self, cluster_id: str, hosts_with_roles, hosts_names: Optional[models.ClusterupdateparamsHostsNames] = None
     ) -> models.cluster.Cluster:
         log.info("Setting roles for hosts %s in cluster %s", hosts_with_roles, cluster_id)
         hosts = models.ClusterUpdateParams(hosts_roles=hosts_with_roles, hosts_names=hosts_names)
         return self.update_cluster(cluster_id=cluster_id, update_params=hosts)
 
+    def infra_env_update_host(self, infra_env_id: str, host_id: str, host_role: str = None, host_name: str = None):
```

Comment:
I'm not sure I understand the function based on the name ``infra_env_update_host``